types/query/collections_pagination.go

```diff
@@ -330,11 +330,11 @@ func encodeCollKey[K, V any, C Collection[K, V]](coll C, key K) ([]byte, error)
 func getCollIter[K, V any, C Collection[K, V]](ctx context.Context, coll C, prefix, start []byte, reverse bool) (collections.Iterator[K, V], error) {
 	// TODO: maybe can be simplified
 	if reverse {
-		var end []byte
-		if prefix != nil {
-			start = storetypes.PrefixEndBytes(append(prefix, start...))
-			end = prefix
-		}
+		// if we are in reverse mode, we need to increase the start key
+		// to include the start key in the iteration.
+		start = storetypes.PrefixEndBytes(append(prefix, start...))
+		end := prefix
```

Comment:
nit: this is not needed, just pass prefix to IterateRaw in the place of end. Although we may want to keep it for readability